### PR TITLE
Implement oauth rules for upgrade

### PR DESCRIPTION
**Pull request by polskikiel**
**Description**

Changes proposed in this pull request:

- Implement oauth rules for upgrade using ORY

**Related issue(s)**

### CODE REVIEW

File: resources/kcp/charts/kyma-environment-broker/templates/api-rule.yaml [hunk 1]
```diff
@@ -21,3 +21,15 @@ spec:
         - handler: oauth2_introspection
           config:
             required_scope: ["cld:read"]
+    - path: /upgrade/.*
```
Comment (piotrmiskiewicz) [inline resources/kcp/charts/kyma-environment-broker/templates/api-rule.yaml:24]: do we have a path`/upgrade` ?

Comment (polskikiel) [reply resources/kcp/charts/kyma-environment-broker/templates/api-rule.yaml:24]: yes, within upgrade API we defined:

- `POST /upgrade/kyma` but in the future we'll also support `/upgrade/cluster`
- `GET /orchestratiions/:id`

**Review by piotrmiskiewicz — APPROVED** [review]

**Review by piotrmiskiewicz — APPROVED** [review]

**Pull request merged**